packages/classes/src/lib/utils/instantiate.util.ts

```diff
@@ -67,7 +67,7 @@ export function instantiate<TModel extends Dictionary<TModel>>(
     }
 
     // if is Date, assign a new Date value if valueAtKey is defined, otherwise, undefined
-    if (isDateConstructor(metaResult)) {
+    if (isDateConstructor(metaResult) && !Array.isArray(valueAtKey)) {
```

Comment:
issue: I think this works but like "unexpectedly works". If `valueAtKey` is a `Date[]` then it will be caught by the next condition (eg: `if Array.isArray(valueAtKey)`) and that condition assumes that `metaResult` has passed all PrimitiveLike (String, Number, Boolean, and Date) model and will try to recursively instantiate in which case `Date` will be used as the model which will fail because there's no metadata for `Date`.

I'd prefer to move the `Array.isArray` check inside this if, and construct a map of Date instead

```ts
value = isDefined(valueAtKey) 
   ? Array.isArray(valueAtKey) 
      ? [...valueAtKey] 
      : new Date(valueAtKey as number) 
   : undefined;
```
something like this? Thoughts?

Reply:
It worked on my implementation for some reason. But, I had to provide a custom mapFrom in the mapping config, which I couldn't before this change as the pre-map value was being set to 'Invalid Date' so I had no source at that point. It's late here, I'll try to get to a fully working solution tomorrow. 

Thanks!